### PR TITLE
Updated iosxe ShowSpanningTreeSummarySchema Optional keys

### PR DESCRIPTION
**Pull request by TestingBytes**
## Description
Proposed fix for iosxe `show spanning-tree summary` parser error described in  https://github.com/CiscoTestAutomation/genieparser/issues/420

Includes change to make keys 'uplink_fast' and 'backbone_fast' optional. 

## Motivation and Context

This will expand support to devices running older versions of ios that do not include this data in the output of `show spanning-tree summary`.

## Impact (If any)
Should be no impact to existing use of the show spanning-tree summary parser.

## Screenshots:
Added new testcase, all tests passing:

image

## Checklist:

- [X] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [X] I have added tests to cover my changes (If applicable).
- [X] All new and existing tests passed.
- [X] All new code passed compilation.

### CODE REVIEW

File: changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+--------------------------------------------------------------------------------
+                                NEW
+--------------------------------------------------------------------------------
+* iosxe
+    * Updated ShowSpanningTreeSummary:
+        * Made 'backbone_fast' and 'uplink_fast' keys optional to resolve https://github.com/CiscoTestAutomation/genieparser/issues/420
```
Comment (LukasMcClelland) [inline changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst:6]: Please change 
```
--------------------------------------------------------------------------------
                                NEW
--------------------------------------------------------------------------------
``` 
to
```
--------------------------------------------------------------------------------
                                FIX
--------------------------------------------------------------------------------
```

File: src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+VLAN0003                     0         0        0          5          5
+VLAN0004                     0         0        0          6          6
+VLAN0005                     0         0        0          5          5
+VLAN0006                     0         0        0          5          5
+VLAN0007                     0         0        0          5          5
+VLAN0008                     0         0        0          5          5
+VLAN0009                     0         0        0          5          5
+---------------------- -------- --------- -------- ---------- ----------
+9 vlans                      0         0        0         46         46
+lab#
```
Comment (LukasMcClelland) [inline src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt:26]: Is `lab#` the device prompt? If so, please remove from output

**Review by LukasMcClelland — CHANGES_REQUESTED** [review]
Thank you for your submission! 

Left 2 small comments

File: src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt (hunk 2 above)
Comment (TestingBytes) [reply src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt]: Good catch. fixed.

File: changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst (hunk 1 above)
Comment (TestingBytes) [reply changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst:6]: Done.

**Comment by TestingBytes** [comment]
I see several checks failed, but it looks like it's because the check has been updated but the pipeline isn't pulling the latest version maybe?

```
File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (genie.libs.parser 21.2 (/home/runner/work/genieparser/genieparser/src), Requirement.parse('genie.libs.parser<21.4.0,>=21.3.0'))
Error: Process completed with exit code 1.
```

https://github.com/CiscoTestAutomation/genieparser/pull/421/checks?check_run_id=2240625317#step:5:84

File: src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt (hunk 2 above)
Comment (LukasMcClelland) [reply src/genie/libs/parser/iosxe/tests/ShowSpanningTreeSummary/cli/equal/golden_output_3_output.txt]: Confirmed the change

File: changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst (hunk 1 above)
Comment (LukasMcClelland) [reply changelog/undistributed/changelog_20210329_iosxe_show_spanning_tree_summary.rst:6]: Confirmed the change

**Comment by LukasMcClelland** [comment]
> 
> 
> I see several checks failed, but it looks like it's because the check has been updated but the pipeline isn't pulling the latest version maybe?
> 
> ```
> File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pkg_resources/__init__.py", line 792, in resolve
>     raise VersionConflict(dist, req).with_context(dependent_req)
> pkg_resources.VersionConflict: (genie.libs.parser 21.2 (/home/runner/work/genieparser/genieparser/src), Requirement.parse('genie.libs.parser<21.4.0,>=21.3.0'))
> Error: Process completed with exit code 1.
> ```
> 
> https://github.com/CiscoTestAutomation/genieparser/pull/421/checks?check_run_id=2240625317#step:5:84

Yes, a new version has been released within the last 48 hours. Your code and PR look good to me. Once the version issues are ironed out and the tests pass again, I'll approve.

**Comment by LukasMcClelland** [comment]
Good news! Version issues have been sorted out. I'll have to ask you to update your fork to get things to line up but after that, tests should pass. I made a fork of your repo and checked out your branch just to check it myself. I followed [this guide](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-an-upstream-repository-into-your-fork) and it worked fine.

Here are the commands I used (but **please make a backup of your work** just in case anything goes awry!):
```
git checkout master
git pull https://github.com/CiscoTestAutomation/genieparser master
git checkout iosxe-showspanningtreesummary
git merge master
```
Then push your updated branch.

**Comment by TestingBytes** [comment]
should be good to go now :)

**Review by LukasMcClelland — APPROVED** [review]

**Comment by LukasMcClelland** [comment]
Approved and ready for merge

**Comment by LukasMcClelland** [comment]
Decided to merge for you, just to avoid any future merge conflicts or issues!

**Comment by TestingBytes** [comment]
@LukasMcClelland Thank you! As a community member, I thought I didn't have write access and had to have someone perform the merge for me.

**Comment by LukasMcClelland** [comment]
I believe you are correct on that one :)

**Pull request merged**